pkg/agent/controller/traceflow/traceflow_controller_test.go

```diff
@@ -740,13 +808,17 @@ func TestSyncTraceflow(t *testing.T) {
 			tfc.crdInformerFactory.Start(stopCh)
 			tfc.crdInformerFactory.WaitForCacheSync(stopCh)
 
-			tfc.runningTraceflows[tt.tf.Status.DataplaneTag] = tt.tfState
+			if tt.existingState != nil {
+				tfc.runningTraceflows[tt.tf.Status.DataplaneTag] = tt.existingState
+			}
```

Comment:
nit: you could just write `tfc.runningTraceflows[tt.tf.Status.DataplaneTag] = tt.existingState` without the if condition

Reply:
The controller assumes value is not empty when the key exists and will access the value's member directly, this would break it.
```
if ok && tfState.name != traceflowName
```

Reply:
oops, yes my comment made no sense...